api-docs/openapi.yaml

```
@@ -0,0 +1,503 @@
+      summary: Search for hadiths using site data
+      parameters:
+        - in: query
+          name: value
+          schema:
+            type: string
+          required: true
+          description: Search query for hadith text
+        - in: query
+          name: page
+          schema:
+            type: integer
+          description: Page number for pagination
+        - in: query
+          name: removehtml
+          schema:
+            type: boolean
+          description: Remove HTML tags from results
+        - in: query
+          name: specialist
+          schema:
+            type: boolean
+          description: Include specialist hadiths
+        - in: query
+          name: xclude
+          schema:
+            type: string
+          description: Words or phrases to exclude from search
+        - in: query
+          name: st
+          schema:
+            type: string
+            enum: [w, a, p]
+          description: Search type (w=all words, a=any word, p=exact phrase)
+        - in: query
+          name: t
+          schema:
+            type: string
+            enum: ['*', '0', '1', '2', '3']
+          description: Search scope
+        - in: query
+          name: d[]
+          schema:
+            type: array
+            items:
+              type: string
+          description: Hadith degree
+        - in: query
+          name: m[]
+          schema:
+            type: array
+            items:
+              type: string
+          description: Muhadith (narrator) IDs
+        - in: query
+          name: s[]
+          schema:
+            type: array
+            items:
+              type: string
+          description: Book IDs
+        - in: query
+          name: rawi[]
+          schema:
+            type: array
+            items:
+              type: string
+          description: Rawi (transmitter) IDs
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SiteHadithSearchResponse'
+
+  /site/hadith/{id}:
+    get:
+      summary: Get a specific hadith by ID from site
+      parameters:
+        - in: path
+          name: id
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SiteSingleHadithResponse'
+
+  /site/hadith/similar/{id}:
+    get:
+      summary: Get similar hadiths by ID
+      parameters:
+        - in: path
+          name: id
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/SiteHadithSearchResponse'
```

@younes-io forget to apply it ?

```suggestion
                $ref: '#/components/schemas/SiteSimilarHadithResponse'
```

---

Sorry, how abt now?